lab-adan/test/company-router-test.js

```diff
@@ -0,0 +1,107 @@
+require('dotenv').config({ path: `${__dirname}/../.test.env` });
+
+const faker = require('faker');
+const expect = require('expect');
+const superagent = require('superagent');
+const server = require('../lib/server.js');
+
+const clearDB = require('./lib/clear-db.js');
+const mockCompany = require('./lib/mock-company.js');
+
+let tempCompany;
+const API_URL = process.env.API_URL;
+
+describe('testing company routes', () => {
+  before(server.start);
+  after(server.stop);
+  afterEach(clearDB);
+
+  describe('testing POST /api/company', () => {
+    let data = {companyName: faker.company.companyName()};
+    console.log('data', data);
+
+    it('should respond with a company', () => {
+      return superagent.post(`${API_URL}/api/company`)
+      .send(data)
+      .then(res => {
+        console.log('data', data);
+        expect(res.status).toEqual(200);
+        expect(res.body.companyName).toEqual(data.companyName);
+        expect(res.body.products).toEqual([]);
+        expect(res.body._id).toExist();
+      });
+    });
+
+    it('should respond with 400 invalid request body', () => {
+      return superagent.post(`${API_URL}/api/company`).send().catch(err => {
+        expect(err.status).toEqual(400);
+      });
+    });
+  });
+
+  describe('testing GET /api/company', () => {
+    it('should respond with a an array of 50 company', () => {
+      return mockCompany.createMany(100)
+      .then(company => {
+        tempCompany = company;
```

Comment:
Take a look back at the module that you're using to set up the sample data. 'company' here is going to be an array not a single value. Below when you try to reference the ._id it's not going to work so you're not going to ever get into the 'then' block of the test.